.taskcluster.yml

```diff
@@ -1,304 +1,282 @@
+policy:
+  pullRequests: public
 tasks:
-  - provisionerId: "{{ taskcluster.docker.provisionerId }}"
-    workerType: "{{ taskcluster.docker.workerType }}"
-    scopes:
-      - secrets:get:repo:github.com/mozilla/balrog:coveralls
-    payload:
-      maxRunTime: 1200
-      image: "mozillareleases/python-test-runner"
-      env:
-        NO_VOLUME_MOUNT: 1
-      command:
-        - "/bin/bash"
-        - "-c"
-        - "git clone $GITHUB_HEAD_REPO_URL balrog && cd balrog && git checkout $GITHUB_HEAD_BRANCH && bash run-tests.sh backend"
-      features:
-        dind: true
-        taskclusterProxy: true
-    extra:
-      github:
-        env: true
-        events:
-          - pull_request.opened
-          - pull_request.synchronize
-          - pull_request.reopened
-          - push
-        branches:
-          - master
-    metadata:
-      name: Balrog back-end tests
-      description: Balrog Python tests
-      owner: "{{ event.head.user.email }}"
-      source: "{{ event.head.repo.url }}"
+  $let:
+    # Github events have this stuff in different places...
+    repo_url: {
+      $if: 'tasks_for == "github-push"',
+      then: '${event.repository.clone_url}',
+      else: {
+        $if: 'tasks_for == "github-pull-request"',
+        then: '${event.pull_request.head.repo.clone_url}',
+        # Assume release event
+        else: '${event.repository.clone_url}',
+      }
+    }
+    head_sha: {
+      $if: 'tasks_for == "github-push"',
+      then: '${event.after}',
+      else: {
+        $if: 'tasks_for == "github-pull-request"',
+        then: '${event.pull_request.head.sha}',
+        # Assume release event
+        else: '${event.release.tag_name}',
+      }
+    }
+    owner_email: {
+      $if: 'tasks_for == "github-push"',
+      then: '${event.pusher.email}',
+      else: {
+        $if: 'tasks_for == "github-pull-request"',
+        then: '${event.pull_request.user.login}@users.noreply.github.com',
+        # Assume release event
+        else: '${event.sender.login}@users.noreply.github.com',
+      }
+    }
+    docker_tags: {
+      $if: 'tasks_for == "github-push"',
+      then: 'latest ${event.after}',
+      else: {
+        $if: 'tasks_for == "github-release"',
+        then: '${event.release.tag_name}',
+        # Unused
+        else: ''
+      }
+    }
+    # jsone doesn't support shortcircuiting evaluation, and we need to use
+    # ${event.release.tag_name} as part of a conditional for release event
+    # tasks, so we need to create this variable to make sure it always
+    # has a value instead of using ${event.release.tag_name} directly.
+    release_tag: {
+      $if: 'tasks_for == "github-release"',
+      then: '${event.release.tag_name}',
+      else: 'null',
+    }
+  in:
+    $flattenDeep:
+      - $if: '(tasks_for == "github-pull-request" && event["action"] in ["opened", "reopened", "synchronize"]) || tasks_for == "github-push"'
+        then:
+          - taskId: {$eval: as_slugid("backend-tests")}
```

Comment:
Why do you need to provide a name/reference for this task?

Reply:
I didn't know it could be omitted :).